arrow-libs/core/arrow-core/src/commonTest/kotlin/arrow/core/ValidatedTest.kt

```diff
@@ -498,5 +498,17 @@ class ValidatedTest : UnitSpec() {
           invalid.bitraverseEither({ it.left() }, { it.right() })
       }
     }
+
+    "andThen should return Valid(result) if f return Valid" {
+      Valid(10).andThen { Valid(it + 5) } shouldBe Valid(15)
+    }
+
+    "andThen should only run f on valid instances " {
+      Invalid(10).andThen { Valid(5) } shouldBe Invalid(10)
+    }
+
+    "andThen should return Invalid(result) if f return Invalid " {
+      Valid(10).andThen { Invalid(5) } shouldBe Invalid(5)
+    }
```

Comment:
Tip:
If you'd want to turn this into property based tests you would simply abstract the two input values:

```
    "andThen should return Valid(result) if f return Valid" {
      checkAll(Arb.int(), Arb.int()) { x, y ->
        Valid(x).andThen { Valid(it + y) } shouldBe Valid(x + y)
      }
    }
```

Reply:
Good suggestions, I'll fix them and push an update